File: src/main/java/de/hsrm/mi/abschlussarbeit/customerRequestProcessingServer/notification/UserNotificationEvent.java
package de.hsrm.mi.abschlussarbeit.customerRequestProcessingServer.notification;

import java.time.Instant;
import java.util.List;

public record UserNotificationEvent(UserNotificationType type, Long processItemId, String processItemTitle, List<Long> userIdsToNotify, String text,
                                    Instant timeStamp, TargetType targetType) {
}

File: frontend/Workstation/src/documentTypes/dtds/UserNotificationEvent.ts
import type {UserNotificationType} from "@/documentTypes/types/UserNotificationType.ts";
import type {TargetType} from "@/documentTypes/types/TargetType.ts";

export interface UserNotificationEvent {
  type: UserNotificationType
  processItemId: number
  processItemTitle: string
  userIdsToNotify: number[]
  text: string
  timeStamp: string
  targetType: TargetType
}
